### DISCUSSION TITLE
OpenDataCam v4 public roadmap

### DISCUSSION THREAD

**Discussion by tdurand** [opendatacam/opendatacam#526] [Announcements]
I took some time to do issue triage and create a milestone for v4 release and I created a [Roadmap on GitHub](https://github.com/orgs/opendatacam/projects/2)

The very short summary of the next version is: OpenDataCam v4 will be a maintenance release focusing on bringing OpenDataCam back on stable ground by

- making sure all documentation is up to date such as https://github.com/opendatacam/opendatacam/issues/503
- fixing all dependency issue such as https://github.com/opendatacam/opendatacam/issues/602
- adding support for current HW such as https://github.com/opendatacam/opendatacam/issues/405
- shipping important bug fixes such as https://github.com/opendatacam/opendatacam/issues/599
- introducing new features that are already working in development such as https://github.com/opendatacam/opendatacam/issues/187

The complete scope and progress of OpenDataCam v4 is managed via the public [Roadmap on GitHub](https://github.com/orgs/opendatacam/projects/2)

*Update 2023-06-29* Edited to update to current v4 scope. @vsaw

**Comment by vsaw** [comment]
How difficult would it be to merge the Android and Jetson forks?

I think it would make all our life's easier if we can avoid maintaining 2 forks.

**Comment by tdurand** [comment]
it is not super simple, diff is documented here: https://github.com/opendatacam/opendatacam-mobile/blob/master/DEVELOPER_NOTES.md#documented-diff-between-main-repo-for-the-nodejs--nextjs-part  .. some stuff would be tricky to support both.. (not impossible though, just takes time to test and dev)

I will not have the time to do it for v4 release.. but I'm happy to keep this "mobile" branch, and document a bit more how we pull the opendatacam code into the android app (do an architecture schema).

My approach is to see if the Android app is of any use for people, and if it gets traction let's later on work on having exactly the same codebase

**Comment by tdurand** [comment]
> I think it would make all our life's easier if we can avoid maintaining 2 forks.

My approach is we don't maintain the Android code, if we want to fix stuff for the Android app, first we do the job to merge the codebase.. right now just consider the https://github.com/opendatacam/opendatacam-mobile as a prototype we don't provide support

**Comment by tdurand** [comment]
Jetpack 4.6 out: https://www.jetsonhacks.com/2021/08/10/jetpack-4-6-production-release/

**Comment by vsaw** [comment]
Development of OpenDataCam v4.0.0 is on the way! You can follow the open Tickets here: https://github.com/opendatacam/opendatacam/projects/3
